x-pack/plugins/fleet/server/services/epm/airgapped.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { appContextService } from '..';
+
+export const airGappedUtils = () => {
+  const config = appContextService.getConfig();
+  const hasRegistryUrls = config?.registryUrl || config?.registryProxyUrl;
+  const isAirGapped = config?.isAirGapped;
+
+  const shouldSkipRegistryRequests = isAirGapped && !hasRegistryUrls;
+
+  return { hasRegistryUrls, isAirGapped, shouldSkipRegistryRequests };
```

Comment:
we should probably create a documentation request to clarify what combination of configs prevents requests to EPR

Reply:
I think we should skip requests when `isAirGapped:true` is set, regardless of whether `registryUrl` is set. Currently if I set both, the requests are still made.

Reply:
> I think we should skip requests when isAirGapped:true is set, regardless of whether registryUrl is set. Currently if I set both, the requests are still made.

This was my first idea too but having an airgapped environment with a local registry is a legit use case that we are already supporting (see [documents](https://www.elastic.co/guide/en/fleet/current/air-gapped.html)). 

If I just block the requests when `isAirGapped` is set to true, I'm going to break this use case for many users, that's why @kpollich [proposed](https://github.com/elastic/kibana/pull/202435#issuecomment-2520454782) to check both conditions. 

> For some reason I don't see all bundled packages, e.g. endpoint which is not installed.
I noticed it too, I'm not sure why. I can open a follow up ticket to check what's going on.

Reply:
I see, it's ok then